The job **`x86_64-msvc`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/10183488254/job/28168550524) [(plain)](https://github.com/rust-lang-ci/rust/commit/e997e9c28a20659ffeaf06f97dae9077ce0bb2fc/checks/28168550524/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Updating files:  99% (48402/48879)
Updating files:  99% (48848/48879)
Updating files: 100% (48879/48879)
Updating files: 100% (48879/48879), done.
branch 'try' set up to track 'origin/try'.
Switched to a new branch 'try'
[command]"C:\Program Files\Git\bin\git.exe" log -1 --format='%H'
'e997e9c28a20659ffeaf06f97dae9077ce0bb2fc'
##[group]Run src/ci/scripts/setup-environment.sh
src/ci/scripts/setup-environment.sh
---
file:.git/config remote.origin.url=https://github.com/rust-lang-ci/rust
file:.git/config remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
file:.git/config gc.auto=0
file:.git/config http.https://github.com/.extraheader=AUTHORIZATION: basic ***
file:.git/config branch.try.remote=origin
file:.git/config branch.try.merge=refs/heads/try
file:.git/config submodule.library/backtrace.url=https://github.com/rust-lang/backtrace-rs.git
file:.git/config submodule.library/stdarch.active=true
file:.git/config submodule.library/stdarch.url=https://github.com/rust-lang/stdarch.git
file:.git/config submodule.src/doc/book.active=true
---
   Compiling clap_complete v4.4.6
   Compiling build_helper v0.1.0 (C:\a\rust\rust\src\tools\build_helper)
   Compiling xz2 v0.1.7
   Compiling sysinfo v0.31.0
error[E0599]: no method named `refresh_cpu` found for struct `sysinfo::System` in the current scope
   |
   |
80 |         state.system_info.refresh_cpu();
   |
help: there is a method `refresh_cpu_all` with a similar name
   |
   |
80 |         state.system_info.refresh_cpu_all();


error[E0599]: no method named `refresh_cpu` found for struct `sysinfo::System` in the current scope
    |
    |
113 |             state.system_info.refresh_cpu();
    |
help: there is a method `refresh_cpu_all` with a similar name
    |
    |
113 |             state.system_info.refresh_cpu_all();


error[E0599]: no method named `refresh_cpu` found for struct `sysinfo::System` in the current scope
    |
    |
151 |         state.system_info.refresh_cpu();
    |
help: there is a method `refresh_cpu_all` with a similar name
    |
    |
151 |         state.system_info.refresh_cpu_all();


error[E0599]: no method named `refresh_cpu` found for struct `sysinfo::System` in the current scope
    |
    |
163 |         system.refresh_cpu();
    |
help: there is a method `refresh_cpu_all` with a similar name
    |
163 |         system.refresh_cpu_all();
---
make: *** [Makefile:94: prepare] Error 1
Command failed. Attempt 2/5:
##[group]Building bootstrap
   Compiling bootstrap v0.0.0 (C:\a\rust\rust\src\bootstrap)
error[E0599]: no method named `refresh_cpu` found for struct `sysinfo::System` in the current scope
   |
   |
80 |         state.system_info.refresh_cpu();
   |
help: there is a method `refresh_cpu_all` with a similar name
   |
   |
80 |         state.system_info.refresh_cpu_all();


error[E0599]: no method named `refresh_cpu` found for struct `sysinfo::System` in the current scope
    |
    |
113 |             state.system_info.refresh_cpu();
    |
help: there is a method `refresh_cpu_all` with a similar name
    |
    |
113 |             state.system_info.refresh_cpu_all();


error[E0599]: no method named `refresh_cpu` found for struct `sysinfo::System` in the current scope
    |
    |
151 |         state.system_info.refresh_cpu();
    |
help: there is a method `refresh_cpu_all` with a similar name
    |
    |
151 |         state.system_info.refresh_cpu_all();


error[E0599]: no method named `refresh_cpu` found for struct `sysinfo::System` in the current scope
    |
    |
163 |         system.refresh_cpu();
    |
help: there is a method `refresh_cpu_all` with a similar name
    |
163 |         system.refresh_cpu_all();
---
make: *** [Makefile:94: prepare] Error 1
Command failed. Attempt 3/5:
##[group]Building bootstrap
   Compiling bootstrap v0.0.0 (C:\a\rust\rust\src\bootstrap)
error[E0599]: no method named `refresh_cpu` found for struct `sysinfo::System` in the current scope
   |
   |
80 |         state.system_info.refresh_cpu();
   |
help: there is a method `refresh_cpu_all` with a similar name
   |
   |
80 |         state.system_info.refresh_cpu_all();


error[E0599]: no method named `refresh_cpu` found for struct `sysinfo::System` in the current scope
    |
    |
113 |             state.system_info.refresh_cpu();
    |
help: there is a method `refresh_cpu_all` with a similar name
    |
    |
113 |             state.system_info.refresh_cpu_all();


error[E0599]: no method named `refresh_cpu` found for struct `sysinfo::System` in the current scope
    |
    |
151 |         state.system_info.refresh_cpu();
    |
help: there is a method `refresh_cpu_all` with a similar name
    |
    |
151 |         state.system_info.refresh_cpu_all();


error[E0599]: no method named `refresh_cpu` found for struct `sysinfo::System` in the current scope
    |
    |
163 |         system.refresh_cpu();
    |
help: there is a method `refresh_cpu_all` with a similar name
    |
163 |         system.refresh_cpu_all();
---
make: *** [Makefile:94: prepare] Error 1
Command failed. Attempt 4/5:
##[group]Building bootstrap
   Compiling bootstrap v0.0.0 (C:\a\rust\rust\src\bootstrap)
error[E0599]: no method named `refresh_cpu` found for struct `sysinfo::System` in the current scope
   |
   |
80 |         state.system_info.refresh_cpu();
   |
help: there is a method `refresh_cpu_all` with a similar name
   |
   |
80 |         state.system_info.refresh_cpu_all();


error[E0599]: no method named `refresh_cpu` found for struct `sysinfo::System` in the current scope
    |
    |
113 |             state.system_info.refresh_cpu();
    |
help: there is a method `refresh_cpu_all` with a similar name
    |
    |
113 |             state.system_info.refresh_cpu_all();


error[E0599]: no method named `refresh_cpu` found for struct `sysinfo::System` in the current scope
    |
    |
151 |         state.system_info.refresh_cpu();
    |
help: there is a method `refresh_cpu_all` with a similar name
    |
    |
151 |         state.system_info.refresh_cpu_all();


error[E0599]: no method named `refresh_cpu` found for struct `sysinfo::System` in the current scope
    |
    |
163 |         system.refresh_cpu();
    |
help: there is a method `refresh_cpu_all` with a similar name
    |
163 |         system.refresh_cpu_all();
---
make: *** [Makefile:94: prepare] Error 1
Command failed. Attempt 5/5:
##[group]Building bootstrap
   Compiling bootstrap v0.0.0 (C:\a\rust\rust\src\bootstrap)
error[E0599]: no method named `refresh_cpu` found for struct `sysinfo::System` in the current scope
   |
   |
80 |         state.system_info.refresh_cpu();
   |
help: there is a method `refresh_cpu_all` with a similar name
   |
   |
80 |         state.system_info.refresh_cpu_all();


error[E0599]: no method named `refresh_cpu` found for struct `sysinfo::System` in the current scope
    |
    |
113 |             state.system_info.refresh_cpu();
    |
help: there is a method `refresh_cpu_all` with a similar name
    |
    |
113 |             state.system_info.refresh_cpu_all();


error[E0599]: no method named `refresh_cpu` found for struct `sysinfo::System` in the current scope
    |
    |
151 |         state.system_info.refresh_cpu();
    |
help: there is a method `refresh_cpu_all` with a similar name
    |
    |
151 |         state.system_info.refresh_cpu_all();


error[E0599]: no method named `refresh_cpu` found for struct `sysinfo::System` in the current scope
    |
    |
163 |         system.refresh_cpu();
    |
help: there is a method `refresh_cpu_all` with a similar name
    |
163 |         system.refresh_cpu_all();
```

</details>

--- a/src/bootstrap/Cargo.toml
+++ b/src/bootstrap/Cargo.toml
@@ -63,7 +63,7 @@ walkdir = "2.4"
 xz2 = "0.1"
 
 # Dependencies needed by the build-metrics feature
-sysinfo = { version = "0.30", default-features = false, optional = true }
+sysinfo = { version = "0.31.2", default-features = false, optional = true, features = ["system"] }
 
 [target.'cfg(windows)'.dependencies.junction]
 version = "1.0.0"

--- a/src/bootstrap/src/utils/metrics.rs
+++ b/src/bootstrap/src/utils/metrics.rs
@@ -13,7 +13,7 @@ use build_helper::metrics::{
     JsonInvocation, JsonInvocationSystemStats, JsonNode, JsonRoot, JsonStepSystemStats, Test,
     TestOutcome, TestSuite, TestSuiteMetadata,
 };
-use sysinfo::System;
+use sysinfo::{CpuRefreshKind, RefreshKind, System};
 
 use crate::core::builder::{Builder, Step};
 use crate::utils::helpers::t;
@@ -55,7 +55,9 @@ impl BuildMetrics {
             finished_steps: Vec::new(),
             running_steps: Vec::new(),
 
-            system_info: System::new(),
+            system_info: System::new_with_specifics(
+                RefreshKind::new().with_cpu(CpuRefreshKind::everything()),
+            ),
             timer_start: None,
             invocation_timer_start: Instant::now(),
             invocation_start: SystemTime::now(),
@@ -77,7 +79,7 @@ impl BuildMetrics {
             self.collect_stats(&mut *state);
         }
 
-        state.system_info.refresh_cpu();
+        state.system_info.refresh_cpu_usage();
         state.timer_start = Some(Instant::now());
 
         state.running_steps.push(StepMetrics {
@@ -110,7 +112,7 @@ impl BuildMetrics {
             state.running_steps.last_mut().unwrap().children.push(step);
 
             // Start collecting again for the parent step.
-            state.system_info.refresh_cpu();
+            state.system_info.refresh_cpu_usage();
             state.timer_start = Some(Instant::now());
         }
     }
@@ -148,7 +150,7 @@ impl BuildMetrics {
         let elapsed = state.timer_start.unwrap().elapsed();
         step.duration_excluding_children_sec += elapsed;
 
-        state.system_info.refresh_cpu();
+        state.system_info.refresh_cpu_usage();
         let cpu = state.system_info.cpus().iter().map(|p| p.cpu_usage()).sum::<f32>();
         step.cpu_usage_time_sec += cpu as f64 / 100.0 * elapsed.as_secs_f64();
     }
@@ -159,8 +161,9 @@ impl BuildMetrics {
 
         let dest = build.out.join("metrics.json");
 
-        let mut system = System::new();
-        system.refresh_cpu();
+        let mut system =
+            System::new_with_specifics(RefreshKind::new().with_cpu(CpuRefreshKind::everything()));
+        system.refresh_cpu_usage();
         system.refresh_memory();
 
         let system_stats = JsonInvocationSystemStats {

--- a/src/tools/opt-dist/Cargo.toml
+++ b/src/tools/opt-dist/Cargo.toml
@@ -10,7 +10,7 @@ log = "0.4"
 anyhow = { version = "1", features = ["backtrace"] }
 humantime = "2"
 humansize = "2"
-sysinfo = { version = "0.30", default-features = false }
+sysinfo = { version = "0.31.2", default-features = false, features = ["disk"] }
 fs_extra = "1"
 camino = "1"
 tar = "0.4"

